### PR TITLE
adds decoding scores to Wav2Vec2DecoderWithLMOutput

### PR DESCRIPTION
**Pull request by arampacha**
# What does this PR do?

- Adds `logit_score` and `lm_score` produced by `BeamSearchDecoderCTC` to `Wav2Vec2DecoderWithLMOutput`. Useful for results analysis and pseudo-labeling for self-training.
- Removes padding from batched logits (as returned by `Trainer.predict`) at `batch_decode`. This prevents producing nonsensical tokens at the end of decoded texts.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
The PR is related to Robust ASR week.
@patrickvonplaten @anton-l

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
I'm fine with adding this actually. Is there a reason why you closed the PR? :-)

**Comment by patrickvonplaten** [comment]
I see a new one was opened: https://github.com/huggingface/transformers/pull/15413/files . never mind
